specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keys.json

```diff
@@ -0,0 +1,588 @@
+        },
+        "message": {
+          "type": "string",
+          "description": "User friendly error message. The message is typically localized and may vary with service version."
+        }
+      },
+      "x-ms-external": true
+    },
+    "Attributes": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Determines whether or not the object is enabled."
+        },
+        "nbf": {
+          "x-ms-client-name": "NotBefore",
+          "type": "integer",
+          "format": "int64",
+          "description": "Not before date in seconds since 1970-01-01T00:00:00Z."
+        },
+        "exp": {
+          "x-ms-client-name": "Expires",
+          "type": "integer",
+          "format": "int64",
+          "description": "Expiry date in seconds since 1970-01-01T00:00:00Z."
+        },
+        "created": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Creation time in seconds since 1970-01-01T00:00:00Z."
+        },
+        "updated": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Last updated time in seconds since 1970-01-01T00:00:00Z."
+        },
+        "recoveryLevel": {
+          "type": "string",
+          "description": "The deletion recovery level currently in effect for the object. If it contains 'Purgeable', then the object can be permanently deleted by a privileged user; otherwise, only the system can purge the object at the end of the retention interval.",
+          "enum": [
+            "Purgeable",
+            "Recoverable+Purgeable",
+            "Recoverable",
+            "Recoverable+ProtectedSubscription"
+          ],
+          "x-ms-enum": {
+            "name": "DeletionRecoveryLevel",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "x-nullable": false
+        }
+      },
+      "description": "The object attributes managed by the Azure Key Vault service."
+    },
+    "KeyProperties": {
+      "properties": {
+        "attributes": {
+          "$ref": "#/definitions/KeyAttributes",
+          "description": "The attributes of the key."
+        },
+        "kty": {
+          "type": "string",
+          "minLength": 1,
+          "description": "The type of the key. For valid values, see JsonWebKeyType.",
+          "enum": [
+            "EC",
+            "EC-HSM",
+            "RSA",
+            "RSA-HSM"
+          ],
+          "x-ms-enum": {
+            "name": "JsonWebKeyType",
+            "modelAsString": true
+          }
+        },
+        "keyOps": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "The permitted JSON web key operations of the key. For more information, see JsonWebKeyOperation.",
+            "enum": [
```

Comment:
"backup" and "restore" not needed here?

Reply:
No, backup and restore aren't JsonWebKeyOperations (although they are key permissions that can be configured in a vault's access policy). I included an additional operation ("import") because it will be allowed for a different upcoming feature.

Compare with the swagger spec from data plane: https://github.com/Azure/azure-rest-api-specs/blob/6b46480ebd9b2f2c7220f2e9a9c6754dd524e2d1/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.0/keyvault.json#L4782-L4800

Screenshot from Azure Portal:
![image](https://user-images.githubusercontent.com/19274229/92023610-d90dab00-ed11-11ea-9441-6b3783d2ef90.png)

Reply:
Thanks. I seemed to have confused key_ops for key permissions.